### ISSUE TITLE
Shared infrastructure for additional RDM-like integrals

### ISSUE THREAD

**Issue by jtkrogel** [QMCPACK/qmcpack#974]
A number of estimators (n(k), 1RDM, EKT, 2RDM) share the need for additional integrals to be performed.  Currently each existing estimator separately performs the ratio evaluations needed for the integration.  

It would be better if a shared facility was created that would generate the relevant sets of integration points (uniform random, random via density, randomly shifted uniform grid, etc), and compute the needed wavefunction ratios in an efficient, batched manner.  All estimators dependent on this information could then be given access and make local use of it without duplicated and divergent effort.

**Comment by prckent** [comment]
Agreed. I might advocate for a single implementation of a "super estimator" to keep the code simple and permit easy reuse. A report from someone indicating that the slowdown from multiple estimators is limiting their science would up-prioritize this work.
